### DISCUSSION TITLE
Cannot log in with did:web

### DISCUSSION THREAD

**Discussion by hugeblank** [likeandscribe/frontpage#241] [Ideas]
**Describe the bug**
When attempting to log in with a handle that's associated with a did:web, the `Application error` screen occurs. Or at least that's what I imagine is happening.

**To Reproduce**
1. Go to https://frontpage.fyi/login
2. Type in a handle associated with a web DID (eg: hugeblank.dev)

**Expected behavior**
A redirect to my pds, then a successful log in.

**Desktop (please complete the following information):**
 - OS: archlinux
 - Browser: firefox
 - Version: 133.0

**Comment by hugeblank** [comment]
@tom-sherman I'm attempting to make a PR for this: using `pnpm run dev` in `packages/frontpage` doesn't work, and I have no idea what the `op` command is in the `dev-1pw` script.

**Comment by hugeblank** [comment]
Regardless, here's what I assume is the fix, completely untested due to the above:

```diff
diff --git a/packages/frontpage/lib/data/atproto/did.ts b/packages/frontpage/lib/data/atproto/did.ts
index 2d4488e..71d3185 100644
--- a/packages/frontpage/lib/data/atproto/did.ts
+++ b/packages/frontpage/lib/data/atproto/did.ts
@@ -16,8 +16,9 @@ export function parseDid(s: string): DID | null {
 }
 
 export const getDidDoc = cache(async (did: DID) => {
-  const url = process.env.PLC_DIRECTORY_URL ?? "https://plc.directory";
-  const response = await fetch(`${url}/${did}`, {
+  const components = did.split(":")
+  const url = components[1] == "web" ? `https://${components[2]}/.well-known/did.json` : `${process.env.PLC_DIRECTORY_URL ?? "https://plc.directory"}/${did}`;
+  const response = await fetch(url, {
     next: {
       // TODO: Also revalidate this when we receive an identity change event
       // That would allow us to extend the revalidation time to 1 day
```

**Comment by tom-sherman** [comment]
I've marked this issue as a feature because it's just something we don't support right now, but may do in the future.

The main problem to solve with did:web is the making sure that once a did:web is deleted it's burned forever, it can't be reused. We may also want to investigate some mitigations for if a domain expires and the DID is taken over.

**Comment by hugeblank** [comment]
don't really understand why a did:web would be a blocker for the former, but for the latter maybe just ask the user to associate an email that's not on the same domain as their did.
